### ISSUE TITLE
OpenGIS EPSG results are not using correct Media Type

### ISSUE THREAD

**Issue by nicholascar** [opengeospatial/NamingAuthority#70]
e.g. <http://www.opengis.net/def/crs/EPSG/0/4326> (WGS 84) returns XML but provides no Media Type in the `Content-Type` header to let the client know this. Full response headers are:

```
HTTP/1.1 200 OK
Date: Mon, 04 Jan 2021 11:32:53 GMT
Server: nginx/1.19.1
Content-Length: 2045
X-Hostname: (null)
Keep-Alive: timeout=5, max=100
Connection: Keep-Alive
```

**Comment by ghobona** [comment]
@pebau Please see the enhancement suggested [above](https://github.com/opengeospatial/NamingAuthority/issues/70#issue-778032590).

**Comment by pebau** [comment]
I see the point. We will phase it into our stack as soon as possible, but:
Are the EPSG resolver and the Australian tool doing the same thing? We might want to stay in sync.

**Comment by ghobona** [comment]
@pebau The server at Jacobs University is indeed returning a content-type header.

<img width="991" alt="JU_Screenshot 2023-10-03 at 13 08 40" src="https://github.com/opengeospatial/NamingAuthority/assets/18121038/931da34c-7054-435a-8d8c-bb67ceba3909">

However, the media type for GML is `application/gml+xml` but the server is returning `text/xml;charset=utf-8`. Please see the IANA registration at https://www.iana.org/assignments/media-types/application/gml+xml

----

@RogerLott The IOGP server is not returning any content-type headers. Please see the screenshot below.

<img width="981" alt="IOGP_Screenshot 2023-10-03 at 13 08 54" src="https://github.com/opengeospatial/NamingAuthority/assets/18121038/9fa27889-6269-4535-9da8-9b06ab6f6200">

**Comment by pebau** [comment]
ok, will have a look.

**Comment by ghobona** [comment]
@bangph I think what we really need is to apply content negotiation. Specifically the `Accept` header as specified in [RFC 7231](https://datatracker.ietf.org/doc/html/rfc7231#section-5.3.2). 

Would it be possible to set the content type using the following logic, described below in pseudocode?

```
if (request.getHeader("Accept").equals("application/gml+xml")) then{
     response.setContentType("application/gml+xml");
}
else {
  response.setContentType("text/xml;charset=utf-8");
}

```

**Comment by bangph** [comment]
@ghobona yes it is possible for the way which you suggested. Default it returns `test/xml` and if a client requests with request header `application/gml+xml` then it returns `application/gml+xml` for it.

**Comment by bangph** [comment]
@ghobona I updated SECORE on our server. Now it works like you suggested (you could close the issue after checking)

default request:

```
curl -I 'https://crs.rasdaman.com/def/crs/OGC/0/CRS84'
HTTP/1.1 200 
Date: Wed, 04 Oct 2023 09:06:42 GMT
Server: Apache/2.4.41 (Ubuntu)
Vary: Origin,Access-Control-Request-Method,Access-Control-Request-Headers
Content-Type: text/xml;charset=utf-8

```

and with request header:

```
curl -H 'Accept: application/gml+xml' -I 'https://crs.rasdaman.com/def/crs/OGC/0/CRS84'
HTTP/1.1 200 
Date: Wed, 04 Oct 2023 09:06:57 GMT
Server: Apache/2.4.41 (Ubuntu)
Vary: Origin,Access-Control-Request-Method,Access-Control-Request-Headers
Content-Type: application/gml+xml
```

**Comment by ghobona** [comment]
Thanks @bangph . I have verified the modification. No further action is required on the Jacobs University server.

`application/gml+xml` media type example below:

![Jacobs_University_CRS_Resolver_GML_handling_1](https://github.com/opengeospatial/NamingAuthority/assets/18121038/e160f27b-038d-4b2e-a698-c6bddee4d3c4)

default media type example below:

![Jacobs_University_CRS_Resolver_GML_handling_2](https://github.com/opengeospatial/NamingAuthority/assets/18121038/8ffe202d-57e2-4f0e-867a-05228321619e)

I will close the GitHub Issue after discussing with the maintainers of the IOGP server.
